src/main/webapp/app/exercises/file-upload/assess/file-upload-assessment-dashboard.component.html

```diff
@@ -62,7 +68,8 @@ <h4 jhiTranslate="artemisApp.fileUploadAssessment.dashboard.heading" [translateV
                         {{ submission.participation.participantName }}
                     </td>
                     <td>
-                        <span *ngIf="submission.latestResult && submission.latestResult?.assessor">{{ submission.latestResult!.assessor!.name }}</span>
+                        <div *ngIf="submission.results.length > 1 && submission.results[0].assessor">{{ submission.results[0].assessor!.name }}</div>
```

Comment:
It seems that before the latesResult could be null since we have this `submission.latestResult && ...` check. Now we don't have this undefined/null check anymore, so if the results of a submission are null, I assume there would be a 'can not read .length of undefined' NullPointerExeption. Is it provided that those values can never be undefined/null?

Reply:
Notice that `submission.latestResult` and `submission.results` are different. `submission.results` is **should** always be defined as it is only an array. If there are no results, it simply is an empty array -> then the latestResult is undefined, logically.
Though I guess you could be correct and the `submission.results` array could be undefined. I added checks for this case.

Reply:
Yeah I wasn't sure about this. So submission obviously is never undefined as it gets initialised with [] in the component. However I don't see any information about submission.results and because the old code also checks for undefined we might also do it just in case :P